include/rviz_visual_tools/rviz_visual_tools.h

```diff
@@ -580,6 +582,16 @@ class RvizVisualTools
    */
   void print();
 
+
+  /**
+   * \brief bounds for generateRandomPose
+   */
+  // 6 x 3 Array
+  // min, max, is_random
+  // if is_random = 0, min is taken as a constant value for that parameter
+  // variables = (x, y, z, elevation, azimuth, rotation angle)
+  Eigen::ArrayXXf random_pose_bounds_;
+
```

Comment:
I'd say you've made an overly-powerful function that can be used in any situation but does not perhaps make life easier for the programmer. Generally I've chosen to sacrifice speed and functionality with ease of use / quick to use in code with these packages. To simplify you could:

- Remove the feature of disable/enable random for each element. If you want an element of a pose to not be random, just set very very close bounds on min/max
- Maybe instead of ArrayXXf use a struct that has member variables ``x_min_``, ``x_max_``, ``elevation_min`` etc
- Do not store that struct in the class since only 1 function uses it - rather, allow it to be optionally passed in to the function. If it is not passed in, populate a default one for the user